app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,57 @@
+class TasksController < ApplicationController
+  before_action :set_task, only: %i[show edit update destroy]
+  before_action :set_project
+  def index
+    @tasks = Task.where(project: @project)
+  end
+
+  def new
+    @task = Task.new
```

Comment:
Do you really need it here?

Reply:
I create Task object here because view need to access errors coming from Task validations.
Later on it passes @task when rendering form inside tasks/new.html.erb